### PR TITLE
feat: use localstorage to persist dark mode setting

### PR DESCRIPTION
**Pull request by urangel**
## Description
Uses local storage to persist dark mode setting.
*Describe your changes in detail*

## Motivation and Context
We want this setting to persist across refreshes and login/logouts.
This PR addresses: [GitHub issue or Jira ticket number]

*Why is this change required? What problem does it solve?*

## How Has This Been Tested?
Unit test was added in store.test.ts
*Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc.*

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

File: packages/javascript/bh-shared-ui/src/components/CreateSAMLProviderForm/CreateSAMLProviderForm.tsx [hunk 1]
```diff
@@ -103,18 +103,21 @@ const CreateSAMLProviderForm: React.FC<{
...
-                                                type='file'
-                                                accept='.xml'
-                                                value={fileValue}
-                                                onChange={(e) => {
-                                                    setFileValue(e.target.value);
-                                                    field.onChange(e.target.files as FileList);
-                                                }}
-                                                onBlur={field.onBlur}
-                                            />
+                                            <>
```
Comment (benwaples) [inline packages/javascript/bh-shared-ui/src/components/CreateSAMLProviderForm/CreateSAMLProviderForm.tsx:106]: from the look so it the fragments are not needed?

Also, no need to explain why this change, I just want to double check that its intentionally in this change set?

**Review by benwaples — APPROVED** [review]

File: packages/javascript/bh-shared-ui/src/components/CreateSAMLProviderForm/CreateSAMLProviderForm.tsx (hunk 1 above)
Comment (urangel) [reply packages/javascript/bh-shared-ui/src/components/CreateSAMLProviderForm/CreateSAMLProviderForm.tsx:106]: Good point! Removing the fragment wrapping. I believe I made this change to patch up the unit test as it was looking for a label element.
